frontend/src/employee-frontend/components/messages/MessageEditor.tsx

```diff
@@ -84,11 +151,39 @@ export default React.memo(function MessageEditor({
     }))
   }, [selectedReceivers])
 
+  // update saving/saved/error status in title with debounce
+  const [titleStatus, setTitleStatus] = useState<string>()
+  const debouncedTitleStatus = useDebounce(titleStatus, 250)
+  useEffect(() => {
+    if (!draftSaveResult) return
+    setTitleStatus(
+      draftSaveResult.isLoading
+        ? `${i18n.common.saving}...`
+        : draftSaveResult.isSuccess
+        ? i18n.common.saved
+        : i18n.common.error.unknown
+    )
+
+    const clearStatus = () => setTitleStatus(undefined)
+    const timeoutHandle = setTimeout(clearStatus, 1500)
+    return () => clearTimeout(timeoutHandle)
+  }, [i18n, draftSaveResult])
+
+  const title =
+    debouncedTitleStatus ||
+    messageBody.title ||
+    i18n.messages.messageEditor.newMessage
+
+  const onCloseHandler = useCallback(() => onClose(!!draftSaveResult), [
```

Comment:
As a user I'd expect this to save the result on close, unless I'd choose to discard the draft. At the moment it doesn't to seem to work that way locally.

Reply:
Close is now disabled while there are unsaved changes (mvp).

Reply:
Draft is now saved on close.